server/models/questions/updateEditorToReader.js

```diff
@@ -0,0 +1,38 @@
+const { pool } = require('../database')
+
+/**
+ *
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {String} param0.editor
+ * @param {Number} param0.questionId
+ * @return {Promise}
+ */
+
+function updateEditorToReader({ username, editor, questionId }) {
+  return new Promise((resolve, reject) => {
+    pool.query(
+      `UPDATE questions_editors q1 SET q1.access = ? 
+            WHERE q1.question_id = (
+                SELECT q1.question_id FROM
+                (SELECT DISTINCT question_id FROM questions_editors q2 WHERE q2.question_id=? AND q2.editor=? AND q2.access=?) AS qid
+            )
+            AND q1.editor=?`,
```

Comment:
1. `DISTINCT` is not required because it will be a single entry returned if all conditions are satisfied.
2. I think `EXISTS` is optimized a little more than the above version.
```suggestion
      `UPDATE questions_editors q1 SET q1.access = ? 
            WHERE 
            EXISTS (
                SELECT 1 FROM
                (SELECT question_id FROM questions_editors q2 WHERE q2.question_id=? AND q2.editor=? AND q2.access=?) AS qid
            )
            AND q1.question_id = ?
            AND q1.editor=?`,
```

Reply:
yeah, this seems better